desktop/release-notes.md

```diff
@@ -24,11 +24,59 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
 
 For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)
 
+## 4.21.0
+
+{% include release-date.html date="2023-06-29" %}
+
+{% include desktop-install.md all=true version="4.21.0" build_path="/" %}
+
+### New
```

Comment:
I think we should add a note about the CLI hints being added in this release (related to https://github.com/docker/compose-cli/pull/2252). And also tell users how to disable them using the env var `DOCKER_CLI_HINTS=false`. cc @eunomie

Reply:
added below! Thanks